### PR TITLE
Fixed several spelling errors under colossalai

### PR DESCRIPTION
**Pull request by digger-yu**
## 📌 Checklist before creating the PR

- [ ] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

Not involved

## 📝 What does this PR do?

Optimization Code
Fixed several spelling errors under colossalai,The code logic is not modified
Note on readme.md file changes
GitHub's image caching mechanism will cache images,Add a random number after the last modified link.
so that every time you visit that link, the contributor's image will be refreshed in real time.

## 💥 Checklist before requesting a review

- [ ] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [x] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

**Comment by digger-yu** [comment]
> 
sorry,I don't understand too well, what I need to do ?

**Comment by binmakeswell** [comment]
> > 
> 
> sorry,I don't understand too well, what I need to do ?

Hi @digger-yu Don't worry. You can update these changes according to above comments at digger-yu:main, then we will review it. Thanks.

**Comment by binmakeswell** [comment]
Hi @digger-yu Thank you for your careful and informative contribution. I have reviewed all the content and think it is pretty nice!
I would suggest you fix the bugs according to the above comments first, and then I can merge this PR.
Because Colossal-AI is in high-speed development, modifying too many files at one time is likely to cause conflicts with other contributors' PRs, causing unnecessary extra work. 
For example, you can split your further PR according to the path, docs/example/colossalai/application/etc. 
Thanks.

**Comment by digger-yu** [comment]
@binmakeswell 
It's an honor to contribute to the open source world.
I'll go through it all as soon as I can, it's almost done.
Most of the changes are actually words in comments.
The individual changes like Usgae::Usage:: are related to the code logic.
But I think this is just to restore the function of its design.

**Comment by binmakeswell** [comment]
image

Hi @digger-yu 
Could you please fix bugs according to comments at the beginning of this PR first, rather than adding new changes?  It may only take a few minutes. You've modified almost a hundred files at once, which is a lot for a PR.
Although these changes aren't related to the code logic, unfortunately, some of them are incorrect, and we should fix them before merging this PR.
Thank you very much.

**Comment by digger-yu** [comment]
@binmakeswell 
Sorry for the trouble you caused
question 1 :Here just add a random number to make sure that each visit is the latest list of images and not cached, it is effective
question 2 :Already revert

**Review by binmakeswell — CHANGES_REQUESTED** [review]

File: colossalai/autochunk/search_chunk.py [hunk 1]
```diff
@@ -16,7 +16,7 @@ class SearchChunk(object):
     This is the core class for AutoChunk.
 
     It defines the framework of the strategy of AutoChunk.
-    Chunks will be selected one by one utill search stops.
+    Chunks will be selected one by one utile search stops.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/autochunk/search_chunk.py]: done

File: docs/source/en/features/1D_tensor_parallel.md [hunk 2]
```diff
@@ -52,7 +52,7 @@ CONFIG = dict(parallel=dict(
 ```
 Then Colossal-AI will automatically apply 1D parallelism to all the layers from `colossalai.nn`.
 
-Let's define a model that consists of a two-layer multi-layer perceptron (MLP) as below.
+Let's define a model that consists of a two-layer multi-layer perception (MLP) as below.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply docs/source/en/features/1D_tensor_parallel.md]: revert

File: docs/source/en/features/2D_tensor_parallel.md [hunk 3]
```diff
@@ -70,7 +70,7 @@ CONFIG = dict(parallel=dict(
 ```
 Then Colossal-AI will automatically apply 2D parallelism to all the layers from `colossalai.nn`.
 
-Let's define a model that consists of a two-layer multi-layer perceptron (MLP) as below.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply docs/source/en/features/2D_tensor_parallel.md]: revert

File: docs/source/en/features/2p5D_tensor_parallel.md [hunk 4]
```diff
@@ -68,7 +68,7 @@ CONFIG = dict(parallel=dict(
 ```
 Then Colossal-AI will automatically apply 2.5D parallelism to all the layers from `colossalai.nn`.
 
-Let's define a model that consists of a two-layer multi-layer perceptron (MLP) as below.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply docs/source/en/features/2p5D_tensor_parallel.md]: revert

File: examples/language/gpt/experiments/auto_parallel/README.md [hunk 5]
```diff
@@ -34,11 +34,11 @@ conda install -c conda-forge coin-or-cbc
...
 ## Dataset
 
-For simplicity, the input data is randonly generated here.
+For simplicity, the input data is randomly generated here.
 
 ## Training
 
 ```bash
-#Run the auto parallel resnet example with 4 GPUs with a dummy dataset.
+#Run the auto parallel resent example with 4 GPUs with a dummy dataset.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply examples/language/gpt/experiments/auto_parallel/README.md]: revert

File: colossalai/autochunk/reorder_graph.py [hunk 6]
```diff
@@ -4,7 +4,7 @@
 
 class ReorderGraph(object):
     """
-    Reorder node list and indice trace list
+    Reorder node list and indices trace list
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/autochunk/reorder_graph.py]: revert

File: colossalai/auto_parallel/passes/runtime_preparation_pass.py [hunk 7]
```diff
@@ -427,9 +427,9 @@ def _shard_param(param, target_sharding_spec):
         if target_sharding_spec.dim_partition_dict != {}:
             origin_sharding_spec = ShardingSpec(device_mesh, param.shape, {})
             setattr(param, 'sharding_spec', origin_sharding_spec)
-            # TODO: build a ColoParamter class to manager the distributed parameters
+            # TODO: build a ColoParameter class to manager the distributed parameters
             # we could use .data here, because all the operations just happen before the real training
-            # loop, so we don't need to track these operations in the autograd graph.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/auto_parallel/passes/runtime_preparation_pass.py:432]: revert

File: colossalai/communication/p2p.py [hunk 8]
```diff
@@ -91,7 +91,7 @@ def _communicate(object_send_next: Union[torch.Tensor, List[torch.Tensor]] = Non
                  dtype: torch.dtype = None,
                  scatter_gather_tensors: bool = False) -> Tuple[Union[torch.Tensor, List[torch.Tensor]]]:
     """
-    Adapted from megatron.p2p_communication.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/communication/p2p.py]: revert

File: colossalai/initialize.py [hunk 9]
```diff
@@ -195,7 +195,7 @@ def launch_from_torch(config: Union[str, Path, Config, Dict],
                       backend: str = 'nccl',
                       seed: int = 1024,
                       verbose: bool = True):
-    """A wrapper for colossalai.launch for torchrun or torch.distributed.launch by reading rank and world size
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/initialize.py]: revert

File: colossalai/nn/layer/moe/routers.py [hunk 10]
```diff
@@ -60,7 +60,7 @@ def pop_routing_loss(self) -> torch.Tensor:
 
 class Top1Router(MoeRouter):
     """Top1 router that returns the dispatch mask [s, e, c] and combine weight [s, e, c]
-    for routing usage. More deailted function can be found in the paper about Switch Transformer
+    for routing usage. More derailed function can be found in the paper about Switch Transformer
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/nn/layer/moe/routers.py]: done

File: colossalai/nn/layer/moe/routers.py [hunk 11]
```diff
@@ -143,7 +143,7 @@ def forward(self, inputs: torch.Tensor, use_kernel: bool = False, ep_group: Opti
 
 class Top2Router(MoeRouter):
     """Top2 router that returns the dispatch mask [s, e, c] and combine weight [s, e, c]
-    for routing usage. More deailted function can be found in the paper about ViT-MoE.
+    for routing usage. More derailed function can be found in the paper about ViT-MoE.
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply colossalai/nn/layer/moe/routers.py]: done

File: README.md [hunk 12]
```diff
@@ -399,7 +399,7 @@ You may contact us or participate in the following ways:
 Thanks so much to all of our amazing contributors!
 
 <a href="https://github.com/hpcaitech/ColossalAI/graphs/contributors">
-  <img src="https://contrib.rocks/image?repo=hpcaitech/ColossalAI"  width="800px"/>
+  <img src="https://contrib.rocks/image?repo=hpcaitech/ColossalAI&r="  width="800px"/>
```
[the remark this answers is not in the document]
Comment (digger-yu) [reply README.md]: revert

**Comment by binmakeswell** [comment]
> @binmakeswell Sorry for the trouble you caused question 1 :it just add a random number to make sure that each visit is the latest list of images and not cached, it is effective question 2 :Already revert

oh, I'm so sorry, I forgot to submit the review comments and you actually cannot read them before. 
Thank you very much for your patience and contribution.

**Review by binmakeswell — APPROVED** [review]

**Review by binmakeswell — APPROVED** [review]

**Pull request merged**